# The Goal is to see how t-statistic works 

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("/Users/meetsudra/Documents/GitHub/mathismust/Titanic-Dataset.csv")
df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

population data 

In [10]:
pop = df["Age"].dropna()
pop

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
885    39.0
886    27.0
887    19.0
889    26.0
890    32.0
Name: Age, Length: 714, dtype: float64

sample data

In [12]:
sample_age = pop.sample(25).values
sample_age

array([30., 49., 29., 36., 30., 26., 50., 25., 45., 31., 28., 30., 29.,
       25., 34., 40.,  9., 28.,  3., 24., 22., 36., 18., 36., 35.])

In [13]:
# Ho = mean age is 35
# Ha = mean age is less then 35  

In [15]:
sample_age.mean()
sample_age.std()

10.441915533081083

1. sample is 25 less then 30 means we cannot apply CLT
2. population mean is 35
3. so our xbar is 29.92 and sample std is 10.44
4. alpha is 0.05 

Main thing 
1. we dont have population std thus we will apply t-test 

In [16]:
# Now since we cant apply CLT lets check whether our sample is following normal distribution or not using SHAPIRO WILK TEST 
from scipy.stats import shapiro 

shapiro = shapiro(sample_age)
print(shapiro)

ShapiroResult(statistic=0.9513218419646562, pvalue=0.2684731376176422)


Since p-value is greater then 0.05 we can say that sample_age is normally distributed 

In [18]:
pop_mean =  35 # given
import scipy.stats as stats
t_statistic, p_value = stats.ttest_1samp(sample_age, pop_mean)

print(t_statistic)
print(p_value/2)  # one-tailed test he left side 



-2.383357316752184
0.012709028816572501


### here p-value is less then 0.05 means we reject the null hypothesis and say that the average age in the titanic data is less then 35 

In [19]:
# lets check kitna mean he 

pop.mean()

29.69911764705882

It verifies that the average age is less then 35